### PR TITLE
[Author] [New] Add FlexSearch.js with git auto-update via single package.json

### PR DESCRIPTION
**Pull request by ts-thomas**
# Pull request template for new library

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/nextapps-de/flexsearch
* Official website (optional, not the repository):
* NPM package url (optional): https://www.npmjs.com/package/flexsearch
* License and its reference: Apache-2.0 https://github.com/nextapps-de/flexsearch/blob/master/LICENSE
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 30
  * Count of stars: 2664
  * Count of forks: 45
* NPM download stats (optional):
  * Downloads in the last day: 
  * Downloads in the last week:
  * Downloads in the last month: 1767

## Essential checklist

* [x] I'm the author of this library
  * [x] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@ts-thomas you can just commit on the branch you sending this pull request, it'll be updated ;)

**Comment by ts-thomas** [comment]
Ok sorry, there comes a message that the forked master branch was too old^^

**Comment by ts-thomas** [comment]
Hopefully now its fine :)

File: ajax/libs/FlexSearch/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "source": "git",
+    "target": "git://github.com/nextapps-de/flexsearch.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "flexsearch.min.js",
+          "flexsearch.compact.js",
+          "flexsearch.light.js",
+          "flexsearch.es5.js"
```
Comment (PeterDaveHello) [inline ajax/libs/FlexSearch/package.json]: Just to make sure, these are all working with browsers, right? Maybe try `"!(flexsearch.node.js)"` here to just ignore the one we don't need here?

**Comment by PeterDaveHello** [comment]
@ts-thomas cool,  I'll help squash the commits into a single one, and take a look later ;)

File: ajax/libs/FlexSearch/package.json (hunk 1 above)
Comment (ts-thomas) [reply ajax/libs/FlexSearch/package.json]: Thanks for the hint. All files are browser compatible. The flexsearch.node.js is also compatible but its just a special version for node.js (without web-worker parts).

**Comment by ts-thomas** [comment]
Squashing would be great, I'm currently just on Github, thanks a lot :)

**Comment by MattIPv4** [comment]
@ts-thomas Is there a default file for this library?

**Comment by ts-thomas** [comment]
Hello, yes it is. The file „flexsearch.min.js“ is the default file.

**Comment by MattIPv4** [comment]
@ts-thomas Could you set that as the `filename` property in the json?

File: ajax/libs/FlexSearch/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "fulltext",
+    "fastest",
+    "contextual",
+    "fuzzy",
+    "matching",
+    "scoring",
+    "indexing",
+    "fieldsearch"
+  ],
+  "filename": "",
```
Comment (PeterDaveHello) [inline ajax/libs/FlexSearch/package.json]: Let's set the default file here.

Comment (ts-thomas) [reply ajax/libs/FlexSearch/package.json]: Done. Could anyone squash the commits please? Thanks.

**Comment by ts-thomas** [comment]
@MattIPv4 Done. Thanks.

File: ajax/libs/FlexSearch/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "fulltext",
+    "fastest",
+    "contextual",
+    "fuzzy",
+    "matching",
+    "scoring",
+    "indexing",
+    "fieldsearch"
+  ],
+  "filename": "dist/flexsearch.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/FlexSearch/package.json]: `dist/` should be removed

**Comment by PeterDaveHello** [comment]
Here we go :tada: 
https://cdnjs.com/libraries/FlexSearch

**Comment by ts-thomas** [comment]
Nice one 👍

**Pull request merged**